src/lib.rs

```diff
@@ -1,4 +1,22 @@
 #![cfg_attr(not(test), no_std)]
+#![warn(
+    missing_copy_implementations,
+    missing_debug_implementations,
+    missing_docs,
+    non_ascii_idents,
+    trivial_casts,
+    unused,
+    unused_qualifications,
+    clippy::expect_used,
+    clippy::unwrap_used
```

Comment:
1. I amit I do not see entirely your point. Depends on your target, but I rather like to avoid crashing completely, and thus avoid unwraps where possible. I understand, that instead you would rather panic on any unexpected state instead, than keep the device working, but I am not convinced this is better.
2. I plan to move `syscalls` to `try_syscalls` too, if that is useful in any way (I have 13 of them, out of 52 total).
3. I wish `try_syscall` panicking would be fixed. Is this not a bug in the framework?

Reply:
1. https://blog.burntsushi.net/unwrap/
2. In opcard I use `syscall` when I know it can't fail
3. https://docs.rs/trussed/latest/src/trussed/client.rs.html#747-758 if the client can't create the value and put it in the interchange, `try_syscall` will panic.

I do think it should instead generate an error integrated with the result.